### PR TITLE
Adding p006, p007, p008 and p009 in C# Softmax Activation, Categorical Cross Entropy Loss and Random Optimization

### PR DESCRIPTION
**Pull request by tanaimarin**
C# version for p006 video
In this version spiral dataset was moved as a separate class from the main code.

### CODE REVIEW

**Comment by daniel-kukiela** [comment]
Will you update the spiral data function with the fix from #145?

**Comment by tanaimarin** [comment]
Yes, for sure.

Tanai

**Comment by tanaimarin** [comment]
updated p006 and added p007 as well.
